### PR TITLE
Make dns stuff inherited from openstack.v2.dns

### PR DESCRIPTION
**Pull request by Polina-Gubina**
Make dns stuff inherited from openstack.v2.dns

### CODE REVIEW

File: otcextensions/sdk/dns/v2/floating_ip.py [hunk 1]
```diff
@@ -9,32 +9,99 @@
 # WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 # License for the specific language governing permissions and limitations
 # under the License.
-from openstack import resource
+import six
```
Comment (gtema) [inline otcextensions/sdk/dns/v2/floating_ip.py]: six was only required for keeping py2 compatibility. Now we do not need it and instead we actually should get rid of it through the code. Please replace this usage with urllib library directly

**Review by gtema — CHANGES_REQUESTED** [review]
hmm, this doesn't really look nice - it introduces too much additional code while we ideally want to reduce it. We should perhaps answer few questions:
- do we need to override find/_get_next_link in every resource class?
- the added functions are anyway implemented in the openstacksdk in the base class
- if we __DO__ need those it would make sense to introduce a MixIn to reduce the code

File: otcextensions/sdk/dns/v2/recordset.py [hunk 2]
```diff
@@ -11,52 +11,55 @@
...
-    #: Timestamp when the zone was last updated
+    #: Timestamp when the zone was last updated.
     updated_at = resource.Body('update_at')
-    #: The id of the Zone which this recordset belongs to
-    zone_id = resource.URI('zone_id')
-    #: The name of the Zone which this recordset belongs to
-    zone_name = resource.Body('zone_name')
+
+    @classmethod
+    def _get_next_link(cls, uri, response, data, marker, limit, total_yielded):
```
Comment (gtema) [inline otcextensions/sdk/dns/v2/recordset.py]: recordset upstream already has this function doing what we need

File: otcextensions/sdk/dns/v2/recordset.py [hunk 3]
```diff
@@ -11,52 +11,55 @@
...
-    status = resource.Body('status')
-    #: Time to live, default 300, available value 300-2147483647 (seconds)
-    ttl = resource.Body('ttl', type=int)
-    #: DNS type of the recordset
-    #: Valid values include `A`, `AAA`, `MX`, `CNAME`, `TXT`, `NS`
-    #: In private zones `SOA`, `SRV`, `PTR`
-    type = resource.Body('type')
-    #: Timestamp when the zone was last updated
+    #: Timestamp when the zone was last updated.
     updated_at = resource.Body('update_at')
```
Comment (gtema) [inline otcextensions/sdk/dns/v2/recordset.py:28]: don't we have y typo here?

File: otcextensions/sdk/dns/v2/_base.py [hunk 4]
```diff
@@ -88,11 +88,11 @@ def _get_next_link(cls, uri, response, data, marker, limit, total_yielded):
         # This prevents duplication of query parameters that with large
         # number of pages result in HTTP 414 error eventually.
         if next_link:
```
Comment (gtema) [inline otcextensions/sdk/dns/v2/_base.py:86]: overall I think we can base from sdk _base: _get_next_link is already defined there. btw, do we still need _base class if we switch to sdk resources?

**Review by gtema — CHANGES_REQUESTED** [review]
please split dns and six changes. For six changes you definitely need to have a releasenotes, cause that means full stop of py2 compatibility (while now we have it untested, without six we do not have it for sure).

File: otcextensions/sdk/dns/v2/recordset.py (hunk 3 above)
Comment (Polina-Gubina) [reply otcextensions/sdk/dns/v2/recordset.py:28]: no, in openstack we have  updated_at = resource.Body('updateD_at'), in otc -   updated_at = resource.Body('update_at')

File: otcextensions/sdk/dns/v2/_base.py (hunk 4 above)
Comment (Polina-Gubina) [reply otcextensions/sdk/dns/v2/_base.py:86]: fixed

File: otcextensions/sdk/dns/v2/recordset.py (hunk 2 above)
Comment (Polina-Gubina) [reply otcextensions/sdk/dns/v2/recordset.py]: fixed

**Review by anton-sidelnikov — APPROVED** [review]

**Review by gtema — APPROVED** [review]

**Pull request merged**